raiden/network/protocol.py

```diff
@@ -208,40 +532,50 @@ def send_async(self, receiver_address, message):
             raise ValueError('Invalid address {}'.format(pex(receiver_address)))
 
         if isinstance(message, Ack):
-            raise ValueError('Do not use send for Ack messages or Errors')
-
-        if len(message.encode()) > self.max_message_size:
-            raise ValueError('message size exceeds the maximum {}'.format(self.max_message_size))
+            raise ValueError('Do not use send for Ack messages')
 
+        # Messages that are not unique per receiver can result in hash
+        # colision, e.g. Secret messages. The hash collision has the undesired
+        # effect of aborting message resubmission once /one/ of the nodes
+        # replied with an Ack, adding the receiver address into the echohash to
+        # avoid these collisions.
         messagedata = message.encode()
-
-        # Adding the receiver address into the echohash to avoid collisions
-        # among different receivers.
-        # (Messages that are not unique per receiver
-        # can result in hash colision, eg. Secret message sent to more than one
-        # node, this hash collision has the undesired effect of aborting
-        # message resubmission once a single node replied with an Ack)
         echohash = sha3(messagedata + receiver_address)
 
-        # Don't add the same message twice into the queue
+        if len(messagedata) > UDP_MAX_MESSAGE_SIZE:
+            raise ValueError(
+                'message size exceeds the maximum {}'.format(UDP_MAX_MESSAGE_SIZE)
+            )
+
+        # All messages must be ordered, but only on a per channel basis, since
+        # each channel correspond to a single token address this with the
```

Comment:
Removed the confusing part, it's actually wrong because there could be multiple Registries and the node_address with the token is not sufficient to uniquely identify the channel.